### PR TITLE
Added cloudwatch decompression and data message extraction

### PR DESCRIPTION
**Pull request by tropnikovvl**
Why this is needed is described in this [documentation](https://docs.aws.amazon.com/firehose/latest/dev/writing-with-cloudwatch-logs.html#writing-with-cloudwatch-logs-decompression).

The main caveat is that the feature cannot be enabled “in a simple way” on an already existing kinesis firehose (This PR will not affect existing instances with default settings in any way). This is described in the [documentation](https://docs.aws.amazon.com/firehose/latest/dev/writing-with-cloudwatch-logs.html#Enabling-disabling-decompression).

In any case, this works great on the new Kinesis Firehose.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -433,6 +433,28 @@ variable "msk_source_connectivity_type" {
 ######
 # S3 Destination Configurations
 ######
+variable "enable_cloudwatch_logs_decompression" {
+  description = "Enables or disables Cloudwatch Logs decompression"
+  type        = bool
+  default     = false
+}
+
+variable "cloudwatch_logs_compression_type" {
```
Comment (fdmsantos) [inline variables.tf:442]: Once it seems the only supported type is GZIP, should we eliminate this variable and use GZIP as hardcoded value in cloudwatch_logs_decompression_processor. IMO, i don't see added-value for this variable. What is your opinion?

**Comment by fdmsantos** [comment]
Thanks for this PR. I added one comment. I unable to test it now, but i will try to test it tomorrow and then approve PR

File: variables.tf (hunk 1 above)
Comment (tropnikovvl) [reply variables.tf]: Done

**Comment by fdmsantos** [comment]
Thanks very much for your collaboration on this repo.

**Comment by fdmsantos** [comment]
This PR is included in version 3.2.0 :tada:

**Pull request merged**